projects/demo-playwright/tests/kit/input/input.spec.ts

```diff
@@ -0,0 +1,145 @@
+        await tuiGoto(
+            page,
+            'components/input/API?tuiTextfieldCleaner=true&tuiTextfieldCustomContent=tuiIconVisaMono&tuiHintContent=Some%20content',
+        );
+
+        await expect(new TuiDocumentationPagePO(page).apiPageExample).toHaveScreenshot(
+            '04-custom-normal-icon-content-cleaner-hint.png',
+        );
+    });
+
+    test('input overflow due to placeholder', async ({page}) => {
+        await tuiGoto(
+            page,
+            'components/input/API?tuiTextfieldIconLeft=tuiIconSearch&pseudoFocus=true&placeholder=Lorem%20ipsum%20dolor%20sit%20amet,%20consectetur%20adipiscing%20elit,%20sed%20do%20eiusmod%20tempor%20incididunt%20ut%20labore',
+        );
+
+        const example = new TuiDocumentationPagePO(page).apiPageExample;
+        const input = example.locator('input[tuiTextfield]');
+
+        await input.focus();
+        await expect(new TuiDocumentationPagePO(page).apiPageExample).toHaveScreenshot(
+            '05-placeholder-will-be-hidden-inside.png',
+        );
+        await input.clear();
+        await input.blur();
+        await expect(example).toHaveScreenshot(
+            '06-placeholder-will-be-hidden-inside.png',
+        );
+    });
+
+    test('character descenders overflow the line', async ({page}) => {
+        await tuiGoto(
+            page,
+            '/components/input/API?&pseudoFocus=true&placeholder=big, placeholder, qwerty, jackson, yellow and more',
+        );
+
+        const example = new TuiDocumentationPagePO(page).apiPageExample;
+        const input = example.locator('input[tuiTextfield]');
+
+        await input.clear();
+        await expect(example).toHaveScreenshot('07-character-descenders.png');
+        await input.focus();
+        await page.keyboard.type(
+            "It has been the industry's standard dummy text ever since the 1500s",
+        );
+        await page.keyboard.press('Enter');
+        await input.blur();
+        await expect(example).toHaveScreenshot('08-character-descenders.png');
+    });
+
+    test('can be horizontally scrolled', async ({page}) => {
+        await tuiGoto(page, '/components/input/API?sandboxWidth=300');
+
+        const example = new TuiDocumentationPagePO(page).apiPageExample;
+        const input = example.locator('input[tuiTextfield]');
+
+        await expect(input).toHaveCSS('text-overflow', 'clip');
+        await input.fill(
+            "You should not set 'text-overflow: ellipsis' for input because it can be horizontally scrolled",
+        );
+        await input.blur();
+        await expect(example).toHaveScreenshot('09-horizontally-scrolled.png');
+        await input.focus();
+        await expect(example).toHaveScreenshot('10-horizontally-scrolled.png');
+        await page.evaluate(() => {
+            const input = document.querySelector('input[tuiTextfield]');
+
+            if (input) {
+                input.scrollLeft = input.clientWidth / 2;
+            }
+        });
```

Comment:
Is it better to use `await mouse.wheel(deltaX, deltaY)` ?

https://playwright.dev/docs/api/class-mouse#mouse-wheel

See this question:
* https://github.com/microsoft/playwright/issues/13189#issuecomment-1083569970

Reply:
so difficult to calculate deltaX, deltaY